### DISCUSSION TITLE
Origin Time

### DISCUSSION THREAD

**Discussion by gilb65** [nikosT/Gisola#38] [General]
Dear all, as far as I understood, both seismograms and Green's functions begin at the event's origin time. This may create issues by the time of using long-period seismograms from stations at short epicentral ranges. Is there a manner to pad with zeroes the initiial part of seismograms and GFs?
thanks,
gilberto 

Gilberto Saccorotti
Istituto Nazionale di Geofisica e Vulcanologia
Pisa, Italy

**Comment by nikosT** [comment]
Dear @gilb65,

At the configuration of Gisola you may specify how far from the epicenter, stations should be chosen. So you could probably add an empirical rule there, like the ones in the [Configuration](https://github.com/nikosT/Gisola/wiki/Configuration).

Perhaps, more seismological info can be given by @esokos

**Comment by esokos** [comment]
Dear @gilb65 ,
this is true, but we haven't implement something in gisola for this case. Mainly because this is a code for using in regional distances and as @nikosT , said user may select the shortest epicentral distance. One trick that could work is to artificially change the OT (making it e.g. 20s earlier, instead of padding with zeros, which is a bit "dangerous". Of course the code should search the centroid not around 0 then but around 20 !! maybe it is not so difficult to implement ?? @nikosT ??  Do you have some specific application for this ? you can always use the manual version of ISOLA for this !

**Reply by gilb65** [reply]
Dear @esokos and @nikosT, thanks for your prompt and exhaustive reply; really appreciated. At a first trial, the manual change of origin time didn't work: I guess that the rule about time shift has to be adjusted consequently, will try that. Another point I have to investigate is whether Axitra accounts for the near-field: as a matter of fact,  if the travel time to a given station is shorter than the lower cutoff period of the filter, that means that wavelengths are larger than hypo distance, i.e., one is in the near field. I'll go through that and keep you posted should I reach any significant conclusion. 
In running Gisola (congrats for the huge work, by the way !) I had two other issues, that I solved but that I signal to you should they be useful for any future development. 
(1) if the --event-xml option is used, the code crashes if the quakeml catalog object also contains the focal mechanism class (as is for M>3.5 events in INGV's catalog) 
http://webservices.ingv.it/fdsnws/event/1/query?eventId=30900271&includeallmagnitudes=true&includeallorigins=true&includearrivals=true&includeallstationsmagnitudes=true
(2) I tried to solve it by downloading a short version of the same catalog:
http://webservices.ingv.it/fdsnws/event/1/query?eventId=30900271

but in this case I got the error **Fatal Error in init: list index out of range**.  The config file is shared by gdrive:
https://drive.google.com/file/d/1cOB9iTCBhgqj6dBP2sGxIHI6H2V24rTs/view?usp=sharing
HOWEVER: I easily solved these issues by directly passing to Gisola the location parameters, and al worked perfectly. So, don't bother any longer, and consider these issues just for your reference. And sorry for the long writing!
Best,
gilberto

**Reply by esokos** [reply]
Axitra should be fine for near field (as far as i know !!:-)

**Reply by nikosT** [reply]
Hi @gilb65 , could you please also attach the log file? Does it crash when it starts reading the data or before storing the data? My intuition to this says at the storing phase. @esokos and @gilb65 The code flow is based on the event's OT. So, if you set a new OT (e.g. OT-10s) in the XML file (let's say), the entire procedure will follow the new OT (i.e. retrieving streams, gf calculations, storing info etc.).

**Comment by gilb65** [comment]
Oh yes, just verified. It claims to perform full wavefield computation.
ciao

Il giorno mer 1 feb 2023 alle ore 09:51 esokos ***@***.***>
ha scritto:

> Axitra should be fine for near field (as far as i know !!:-)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/38#discussioncomment-4838296>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU73NIS5YZV4EAFXJAJUOTTWVIPXZANCNFSM6AAAAAAUM2AKKU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
**************************************************************
 Gilberto Saccorotti
 Istituto Nazionale di Geofisica e Vulcanologia - Sezione di Pisa
 Via C. Battisti, 53 - 56125 PISA (I)
 Tel +39 050 8311960
  Skype: g.saccorotti
**************************************************************
